src/docs/development/accessibility-and-localization/internationalization.md

```diff
@@ -101,8 +118,135 @@ More information about these app properties, the types they
+  flutter_localizations:
+    sdk: flutter
+  intl: ^0.17.0-nullsafety.2 # Add this line
+
+# The rest of the pubspec file...
+
+# The following section is specific to Flutter.
+flutter:
+  generate: true # Add this line
+```
+
+Once that is complete, add a new yaml file to the root directory
+of the Flutter project called `l10n.yaml`. This file will determine
+the configuration of your localization generation. Define it as such:
+
+```yaml
+arb-dir: lib/l10n
+template-arb-file: app_en.arb
+output-localization-file: app_localizations.dart
+```
+
+The options above basically mean that the input files for
+the localizations tool will be defined at `${FLUTTER_PROJECT}/lib/l10n`,
+the `app_en.arb` file will be the template file that the tool will use,
+and `app_localizations.dart` will the name of the output base localizations
+file.
+
+Next, we will need to define `app_en.arb` in `${FLUTTER_PROJECT}/lib/l10n`,
+which will contain the template locale's messages. We will use English as
+the template for this example. The locale is inferred by the tool from the
+filename.
+
+```json
+{
+  "helloWorld": "Hello World!",
+  "@helloWorld": {
+    "description": "The conventional newborn programmer greeting"
+  }
+}
+```
+
+Then, let's define an `app_es.arb` in the same directory to contain
+Spanish translations for the message defined above:
+
+```json
+{
+  "helloWorld": "Hola Mundo!"
+}
+```
+
+At this point, try running your Flutter application. If done correctly,
+you should see some generated filed in
+`${FLUTTER_PROJECT}/flutter_gen/gen_l10n`. At this point, you should be
+able to use the generated localizations code in your Flutter application.
+
+You can test this out in application itself as follows:
+<!-- skip -->
+```dart
+import 'package:flutter_localizations/flutter_localizations.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart'; // Add this line
+
+// ...
+
+// Use AppLocalizations anywhere in your app. Here, the translated message
+// is used in a Text widget.
+Widget build(BuildContext context) {
+  // ...
+  return Text(AppLocalizations.of(context).helloWorld);
+}
+```
+
+This should generate a Text widget that displays "Hello World!" if the target
+device's locale is set to English, and "Hola Mundo!" if the target device's
+locale is set to Spanish. In the arb files, the key of each entry will be used
+as the getter's/method's name, while the value of that entry contains the
+localized message.
+
+For more information about the tool, such as dealing with DateTime, handling
+plurals, see flutter.dev/go/i18n-user-guide for a deeper dive.
+
+<a name="ios-specifics"></a>
+### Localizing for iOS: Updating the iOS app bundle
+
+iOS applications define key application metadata,
+including supported locales, in an `Info.plist` file
+that is built into the application bundle.
+To configure the locales supported by your app,
+you'll need to edit this file.
+
+First, open your project's `ios/Runner.xcworkspace` Xcode
```

Comment:
=>
1. Open your project's `ios/Runner.xcworkspace` Xcode file.

2. In the **Project Navigator**, open the `Info.plist` file under
    the `Runner` project's `Runner` folder.

3. Select the **Information Property List** item,
    then select **Add Item** from the **Editor** menu,
    and then select **Localizations** from the pop-up menu.

4. Select and expand the newly-created `Localizations` item.
    For each locale your application supports,
    add a new item and select the locale you wish to add
    from the pop-up menu in the **Value** field.
    This list should be consistent with the languages listed
    in the [`supportedLocales`][] parameter.

5. Once all supported locales have been added, save the file.

---

Shams' question:
- Are these instructions the same for Cupertino AND Material apps on iOS?
- Also, I'd restructure this a bit. What I'd like to see:

# Configuration instructions for Material apps on Android
# Configuration instructions for Cupertino/Material apps on iOS. (I'm not sure how this might differ...)
# Testing instructions for all apps, regardless of iOS/Android.

As it is now, the testing instructions are only in that first set.
Also, what about web and desktop? I assume those will be handled in a different way, but add a note to the top of this page stating that this is for mobile only.

Reply:
> Are these instructions the same for Cupertino AND Material apps on iOS?

Yes, they are the same.

> Also, I'd restructure this a bit. What I'd like to see:

- For Material apps on Android, nothing else needs to be done. It is only extra complex for iOS apps.
- Cupertino/Material apps on iOS is the same, as mentioned before, so there'd be one overarching section just for iOS
- For testing instructions for all apps, could you further elaborate? Is this just additional instructions for every other section other than the one I added to verify that everything is working as intended?
- I'm actually not very sure about web and desktop as I haven't been testing with them. I'll add a note to the top of the page that this is for mobile only.